packages/help-center/src/components/help-center.tsx

```diff
@@ -2,51 +2,136 @@
-	const { setUnreadCount } = useDispatch( HELP_CENTER_STORE );
 	const { setSite } = useDispatch( HELP_CENTER_STORE );
+	const { setShowMessagingLauncher } = useDispatch( HELP_CENTER_STORE );
+	const { setShowMessagingWidget } = useDispatch( HELP_CENTER_STORE );
+	const { setThirdPartyCookiesAllowed } = useDispatch( HELP_CENTER_STORE );
 
-	const { show, isMinimized } = useSelect(
-		( select ) => ( {
-			isMinimized: ( select( HELP_CENTER_STORE ) as HelpCenterSelect ).getIsMinimized(),
-			show: ( select( HELP_CENTER_STORE ) as HelpCenterSelect ).isHelpCenterShown(),
-		} ),
-		[]
-	);
+	const { status: zendeskStatus } = useZendeskConfig( Boolean( chatStatus?.is_user_eligible ) );
+	useEffect( () => {
+		if ( zendeskStatus === 'success' ) {
+			setThirdPartyCookiesAllowed( true );
+		} else if ( zendeskStatus === 'error' ) {
+			setThirdPartyCookiesAllowed( false );
+		}
+	}, [ setThirdPartyCookiesAllowed, zendeskStatus ] );
+
+	useEffect( () => {
+		if ( ! chatStatus?.is_user_eligible ) {
+			return;
+		}
+
+		const zendeskKey: string | false = config( 'zendesk_support_chat_key' );
+		if ( ! zendeskKey ) {
+			return;
+		}
+
+		if ( document.getElementById( ZENDESK_SCRIPT_ID ) ) {
+			return;
+		}
+
+		function setUpMessagingEventHandlers() {
+			if ( typeof window.zE !== 'function' ) {
+				return;
+			}
+
+			window.zE( 'messenger', 'hide' );
+
+			window.zE( 'messenger:on', 'open', function () {
+				setShowMessagingWidget( true );
+			} );
+			window.zE( 'messenger:on', 'close', function () {
+				setShowMessagingWidget( false );
+			} );
+			window.zE( 'messenger:on', 'unreadMessages', function ( count ) {
+				if ( Number( count ) > 0 ) {
+					setShowMessagingLauncher( true );
+				}
+			} );
+		}
 
-	const { unreadCount, closeChat } = useHCWindowCommunicator( isMinimized || ! show );
+		loadScript(
+			'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( zendeskKey ),
+			setUpMessagingEventHandlers,
+			{ id: ZENDESK_SCRIPT_ID }
+		);
+	}, [ setShowMessagingLauncher, setShowMessagingWidget, chatStatus ] );
 
+	const { data: messagingAuth } = useMessagingAuth( Boolean( chatStatus?.is_user_eligible ) );
 	useEffect( () => {
-		setUnreadCount( unreadCount );
-	}, [ unreadCount, setUnreadCount ] );
+		const jwt = messagingAuth?.user.jwt;
+		if ( typeof window.zE !== 'function' || ! jwt ) {
+			return;
+		}
+
+		window.zE( 'messenger', 'loginUser', function ( callback ) {
+			callback( jwt );
+		} );
+	}, [ messagingAuth ] );
+
+	const { showMessagingLauncher, showMessagingWidget } = useSelect( ( select ) => {
+		const helpCenterSelect: HelpCenterSelect = select( HELP_CENTER_STORE );
+		return {
+			showMessagingLauncher: helpCenterSelect.isMessagingLauncherShown(),
+			showMessagingWidget: helpCenterSelect.isMessagingWidgetShown(),
+		};
+	}, [] );
+
+	useEffect( () => {
+		if ( typeof window.zE !== 'function' ) {
+			return;
+		}
+		if ( showMessagingLauncher ) {
```

Comment:
I would make `showMessagingLauncher` a string instead of a boolean and just pass it around.

Reply:
I'm not sure I understand - could you clarify what are the benefits of it being a string and how that would make it easier to pass it around?

Reply:
This `if` is basically converting `true` to `'show'` and `false` to `'hide'`. I prefer 

```js
window.zE( 'messenger', showMessagingLauncher )
```

It's inspired by this https://kentcdodds.com/blog/stop-using-isloading-booleans